tests/test_views.py

```diff
@@ -570,3 +572,53 @@ def test_successful_manager_removal(self, pi, pi_client, pi_group, mailoutbox):
         assert member.email in email.body
         assert member.get_full_name() in email.body
         assert pi.get_full_name() in email.body
+
+
+class TestGroupMembershipExtendView(LoginRequiredMixin):
+    """Tests for the group membership extend view."""
+
+    def _get_url(self, group_membership_pk=1):
+        return reverse(
+            "imperial_coldfront_plugin:extend_membership", args=[group_membership_pk]
+        )
+
+    def test_not_group_owner(
+        self, research_group_factory, auth_client_factory, user_client, pi_group
+    ):
+        """Test non group owner or manager cannot access the view."""
+        group, memberships = research_group_factory(number_of_members=1)
+        client = auth_client_factory(group.owner)
+        response = client.get(self._get_url())
+        assert response.status_code == HTTPStatus.FORBIDDEN
+        assert response.content == b"Permission denied"
+
+    def test_group_owner(self, pi_client, pi_group):
+        """Test that the group owner can extend a group membership."""
+        group_membership = pi_group.groupmembership_set.first()
+        response = pi_client.get(self._get_url(group_membership.pk))
+        assert response.status_code == HTTPStatus.OK
+        assert isinstance(response.context["form"], GroupMembershipExtendForm)
```

Comment:
Need to also check that the group membership is included in the context as this is currently missing which causes info to be missing when the page is rendered.